solutions/USACO dec15-breed-counting.mdx

```diff
@@ -13,37 +13,43 @@ author: Óscar Garries (C++), Owen Wang (Java)
 ## Implementation
 
 ```cpp
-#include <bits/stdc++.h>
+#include <iostream>
 
-using namespace std;
+using std::cin;
+using std::cout;
 
 int main() {
 	freopen("bcount.in", "r", stdin);
 	freopen("bcount.out", "w", stdout);
 
-	int n, q;
-	cin >> n >> q;
-	int h[n + 1], g[n + 1], j[n + 1];
-
-	for (int i = 0; i < n; i++) {
-		h[i + 1] = h[i];
-		g[i + 1] = g[i];
-		j[i + 1] = j[i];
-		int id;
-		cin >> id;
-
-		if (id == 1) h[i + 1]++;
-		else if (id == 2) g[i + 1]++;
-		else j[i + 1]++;
+	int cow_num;
+	int query_num;
+	cin >> cow_num >> query_num;
+
+	int holsteins[cow_num + 1];
+	int guernseys[cow_num + 1];
+	int jerseys[cow_num + 1];
```

Comment:
wait lol don't these need to be initialized to zero (at least for the zeroeth index)?

Reply:
default value's 0 for java arrays

Reply:
this isn't java...

Reply:
also variable length arrays are nonstandard - could you change them to constant size or use vectors?